### PR TITLE
Adds an area flag to prevent turf destruction

### PR DESCRIPTION
**Pull request by RustingWithYou**
Adds AREA_FLAG_INDESTRUCTIBLE_TURFS, which prevents turfs in an area from being destroyed by explosions or fully deconstructed. This allows for fake planet away sites such as Point Verdant and the blue river to be safe from being somehow vented.

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

File: code/game/turfs/simulated/floor_acts.dm [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
-				if(2)
-					src.ChangeTurf(baseturf)
-				if(3)
-					if(prob(33)) new /obj/item/stack/material/steel(src)
-					if(prob(80))
-						src.break_tile_to_plating()
-					else
-						src.break_tile()
+	var/area/A = get_area(src)
+	if(A.area_flags & AREA_FLAG_INDESTRUCTIBLE_TURFS)
```
Comment (NonQueueingMatt) [inline code/game/turfs/simulated/floor_acts.dm:4]: check that the area exists too

**Review by NonQueueingMatt — APPROVED** [review]

File: code/game/turfs/simulated/floor_acts.dm [hunk 2]
```diff
@@ -1,26 +1,35 @@
...
-				if(2)
-					src.ChangeTurf(baseturf)
-				if(3)
-					if(prob(33)) new /obj/item/stack/material/steel(src)
-					if(prob(80))
-						src.break_tile_to_plating()
-					else
-						src.break_tile()
+	var/area/A = get_area(src)
+	if(A && A.area_flags & AREA_FLAG_INDESTRUCTIBLE_TURFS)
```
Comment (Geevies) [inline code/game/turfs/simulated/floor_acts.dm:4]:
```suggestion
	if(A && (A.area_flags & AREA_FLAG_INDESTRUCTIBLE_TURFS))
```

File: code/game/turfs/simulated/floor_attackby.dm [hunk 3]
```diff
@@ -89,6 +89,9 @@
 				return
 		// Repairs and deconstruction
 		else if(attacking_item.iscrowbar())
+			var/area/A = get_area(src)
+			if(A && A.area_flags & AREA_FLAG_INDESTRUCTIBLE_TURFS)
```
Comment (Geevies) [inline code/game/turfs/simulated/floor_attackby.dm:93]:
```suggestion
			if(A && (A.area_flags & AREA_FLAG_INDESTRUCTIBLE_TURFS))
```

**Review by Geevies — APPROVED** [review]

**Pull request merged**